### PR TITLE
feat: add confirm step (modal) before deleting a secret

### PR DESCRIPTION
**Pull request by LemmyMwaura**
# Description 📣

Adds a confirm step (modal) before deleting a secret as referenced in this discussion -> #2154

Infisical-delete-secret-modal

## Type ✨

- [x] New feature

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

File: frontend/src/views/SecretOverviewPage/components/ManageSecretModals/DeleteSecretModal.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { type FC } from "react"
+import { Modal, ModalContent, Button } from "@app/components/v2";
+
+type Props = {
+  isModalOpen: boolean;
+  handleDeleteSecret: () => void;
+  toggleModal: () => void;
+};
+
+export const DeleteSecretModal: FC<Props> = ( { isModalOpen, toggleModal, handleDeleteSecret }) => {
```
Comment (maidul98) [inline frontend/src/views/SecretOverviewPage/components/ManageSecretModals/DeleteSecretModal.tsx:10]: Can you reuse the existing one we already have when we go to delete a secret inside of the dashbaord?
CleanShot 2024-07-26 at 13 51 08@2x

Comment (LemmyMwaura) [reply frontend/src/views/SecretOverviewPage/components/ManageSecretModals/DeleteSecretModal.tsx]: I've updated it.

File: frontend/src/views/SecretOverviewPage/components/SelectionPanel/SelectionPanel.tsx [hunk 2]
```diff
@@ -68,6 +69,16 @@ export const SelectionPanel = ({
     )
   );
 
+  const getDeleteModalTitle = () => {
+    if (selectedFolderCount > 0 && selectedKeysCount > 0) {
+      return "Do you want to delete the selected secrets and folders across envs?";
+    } else if (selectedFolderCount > 0 && selectedKeysCount === 0) {
+      return "Do you want to delete the selected folders across envs?";
```
Comment (maidul98) [inline frontend/src/views/SecretOverviewPage/components/SelectionPanel/SelectionPanel.tsx:76]: change `envs` to environments for all of the tests

**Review by maidul98 — CHANGES_REQUESTED** [review]
Looks good, just one more comment

File: frontend/src/views/SecretOverviewPage/components/SelectionPanel/SelectionPanel.tsx (hunk 2 above)
Comment (LemmyMwaura) [reply frontend/src/views/SecretOverviewPage/components/SelectionPanel/SelectionPanel.tsx]: made the update

**Review by maidul98 — APPROVED** [review]
awesome!

**Pull request merged**